src/AuthButton.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import {
+  BrowserRouter as Router,
+  Route,
+  Link,
+  Redirect,
+  withRouter
+} from 'react-router-dom';
+import auth from './services/awsAuth'
+
+const AuthButton = withRouter(({ history }) => (
+  auth.isAuthenticated() ? (
+    <p>
+      Welcome! <button onClick={() => {
+        auth.signout(() => history.push('/'))
+      }}>Sign out</button>
+    </p>
+  ) : (
+    <p>You are not logged in.</p>
+  )
+));
+
+export default AuthButton;
```

Comment:
<img width="170" alt="screen shot 2017-06-02 at 12 16 27 pm" src="https://cloud.githubusercontent.com/assets/6384156/26737359/580c4884-478f-11e7-85a6-105eb4b8ceda.png">

It hasn't gotten to a point where I'm having trouble managing it yet. I figure once we add some more stuff it'll become a more pressing issue.